addons/web/static/tests/webclient/actions/embedded_action.test.js

```diff
@@ -332,6 +335,60 @@ test("a view coming from a embedded can be saved in the embedded actions", async
     });
 });
 
+test("a view coming from a embedded with python_method can be saved in the embedded actions", async () => {
+    onRpc("create", ({ args }) => {
+        const values = args[0][0];
+        expect(values.name).toBe("Custom Embedded Action 3");
+        expect(values.python_method).toBe("do_python_method");
+        expect(values).not.toInclude("action_id");
+        return [4, values.name]; // Fake new embedded action id
+    });
+    onRpc("create_or_replace", ({ args }) => {
+        expect(args[0].domain).toBe(`[["name", "=", "Applejack"]]`);
+        expect(args[0].embedded_action_id).toBe(4);
+        expect(args[0].user_id).toBe(false);
+        return 5; // Fake new filter id
+    });
+    onRpc("do_python_method", () => {
+        return {
+            id: 4,
+            name: "Favorite Ponies from python action",
+            res_model: "pony",
+            type: "ir.actions.act_window",
+            views: [[false, "list"], [false, "kanban"]],
+        };
+    });
```

Comment:
can't we combine all these to one rpc method and check for the method called kind of like how it's done [here](https://github.com/odoo/odoo/blob/master/addons/project/static/tests/project_task_subtask.test.js#L214-L229)

Reply:
Yep, indeed, I agree with you. 🤔 

TBH, I did not really pay to attention to refactor or merge the tests to not loose the readability. Moreover, we plan to remove `python_method` field in embedded action model because the python method used in the embedded actions could be replaced by a server action.

See <https://www.odoo.com/odoo/project/4105/tasks/4184991>

So, I guess it is okay to keep that duplication only in that version. 🤷🏼‍♂️ 

What do you think? :)

Reply:
Why not. 🙂

Reply:
The funny thing is we will have more lines if we combine all in one rpc function. 😄

Reply:
I pushed it in separate commit and we will see which one we prefer. 🙂

Reply:
personally I tend to prefer the one method + if statement approach but we can leave it as is and decide on the final review

Reply:
Okay let's let like this then. 😁